### PR TITLE
[RELEASE][VERSIONS][DOCS] - bump versions to 5.4.3, document release process

### PR DESCRIPTION
**Pull request by mikehardy**
This one needs more than review, it needs collaboration or I'm lost :-) - you might want to do this (assuming my commands are correct, in an attempt to make this copy-pastable):

```shell
git remote add mikehardy git@github.com:mikehardy/react-native-firebase.git
git fetch mikehardy
git co mikehardy/release-process-5.4.3
```

...so it is easy to edit and just type away then push to this branch in my repo, I'm fine rebasing etc to collaborate and that way you/we just type once?

The idea is:

* bump the versions as requested
* figure out how to actually turn the crank on this thing

I did not edit package.json or anything as I'm not sure what release machinery you've already built

---

Think `react-native-firebase` is great? Please consider supporting the project with any of the below:

- 👉 Donate via [Open Collective](https://opencollective.com/react-native-firebase/donate)
- 👉 Follow [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter
- 👉 Star this repo on GitHub ⭐️
- 👉 Contribute; see our [contributing guide](/CONTRIBUTING.md)

### CODE REVIEW

**Comment by Salakar** [comment]
The actual package and release part is just the following, run on the v5.x.x branch:

```
npm version 5.4.3 # this will make a git tag and also update package json version

npm publish

# add and commit the changes, then push back to v5.x.x:
git push --follow-tags
```

And that's it, there's a pre-publish hook/npm script that takes care of the build etc

----

That's the package and publish part anyway, for  the rest I'll take a look and update this tomorrow if I can, it's late here.

**Comment by Salakar** [comment]
And super woooop for getting this documented 🔥

**Comment by mikehardy** [comment]
Okay - I integrated  your changes, and I also attempted some of the steps (`npm prepublishOnly` that will execute, and fuzzbusted them.

I think this is ready to go then, the branch itself is ready, and if merged contains all the necessary release items, so it would just be following the steps

Please triple-check and if it looks good, approve+merge and then you or I can try following the steps to launch 5.4.3? :crossed_fingers:

File: RELEASE_PROCESS.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+- Make sure there is an entry in docs for the new release: <https://github.com/invertase/react-native-firebase-docs/tree/v5.x.x/docs/releases>
+- Traditionally the new minor versions get a new page (e.g. v5.4.x.md)
+
+## SDK Version numbers
+
+1. Update the firebase registration version numbers to match the next version number
+   - iOS here: <https://github.com/invertase/react-native-firebase/blob/v5.x.x/ios/RNFirebase/RNFirebase.m#L20>
+   - Android here: <https://github.com/invertase/react-native-firebase/blob/v5.x.x/android/src/main/java/io/invertase/firebase/ReactNativeFirebaseAppRegistrar.java#L36>
+1. Update package.json (need guidance here)
```
Comment (Salakar) [inline RELEASE_PROCESS.md]:
```suggestion
```

File: RELEASE_PROCESS.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+1. Create git tag and alter package.json with `npm version 5.x.x` (e.g. `npm version 5.4.3`)
+1. Publish with `npm publish`
+1. Commit changes with `git add -A && git commit -a`
+1. Push changes back to repo with `git push --follow-tags upstream` (substitute your remote ref for `upstream` if the main repository has a different reference name in your local repo)
+1. For new minor versions make an entry github releases page pointing to the new v5.5.x (or v5.6.x etc) releases page on rnfirebase.io
+
+## Release notification
+
+1. discord announcements: invertase / react-native?
+1. twitter?
```
Comment (Salakar) [inline RELEASE_PROCESS.md]: I can do this on the RNFB account and tag you in it =]

File: RELEASE_PROCESS.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+1. Create git tag and alter package.json with `npm version 5.x.x` (e.g. `npm version 5.4.3`)
+1. Publish with `npm publish`
+1. Commit changes with `git add -A && git commit -a`
+1. Push changes back to repo with `git push --follow-tags upstream` (substitute your remote ref for `upstream` if the main repository has a different reference name in your local repo)
+1. For new minor versions make an entry github releases page pointing to the new v5.5.x (or v5.6.x etc) releases page on rnfirebase.io
+
+## Release notification
+
+1. discord announcements: invertase / react-native?
```
Comment (Salakar) [inline RELEASE_PROCESS.md]:
```suggestion
1. Discord announcement: Invertase OSS > React Native Firebase > Announcements (use `@here` in the announcement to notify anyone watching the channel)
```

**Pull request merged**